mlflow/transformers.py

```diff
@@ -0,0 +1,793 @@
+_CARD_TEXT_FILE_NAME = "model_card_text.txt"
+_CARD_DATA_FILE_NAME = "model_card_data.yaml"
+_TASK_KEY = "task"
+_LOGGED_TYPE_KEY = "logged_type"
+_INSTANCE_TYPE_KEY = "instance_type"
+_PIPELINE_MODEL_TYPE_KEY = "pipeline_model_type"
+_MODEL_PATH_OR_NAME_KEY = "source_model_name"
+
+_logger = logging.getLogger(__name__)
+
+
+def _model_engine_type(model) -> List[str]:
+    """
+    Determines which pip libraries should be included based on the base model engine
+    type.
+    In the ``transformers`` library, all TensorFlow versions of specific model implementations have
+    their class names begin with "TF" (for instance, "TFMobileBertForSequenceClassification" is the
+    TensorFlow version of the MobileBertForSequenceClassification model) while the
+    PyTorch version of the same architecture omits the "TF" ("MobileBertForSequenceClassification").
+
+    :param model: The model instance to be saved in order to provide the required underlying
+                  deep learning execution framework dependency requirements.
+    :return: A string representing the engine type of the model being used (one of 'tf', 'flax', or
+             'torch').
+    """
+    engine = _get_engine_type(model)
+    if engine == "torch":
+        return ["torch", "torchvision"]
+    else:
+        return [engine]
+
+
+def get_default_pip_requirements(model) -> List[str]:
+    """
+    :param model: The model instance to be saved in order to provide the required underlying
+                  deep learning execution framework dependency requirements. Note that this must
+                  be the actual model instance and not a Pipeline.
+    :return: A list of default pip requirements for MLflow Models that have been produced with the
+             ``transformers`` flavor. Calls to :py:func:`save_model()` and :py:func:`log_model()`
+             produce a pip environment that contain these requirements at a minimum.
+    """
+    from transformers import TFPreTrainedModel, FlaxPreTrainedModel, PreTrainedModel
+
+    if not isinstance(model, (TFPreTrainedModel, FlaxPreTrainedModel, PreTrainedModel)):
+        raise MlflowException(
+            "The supplied model type is unsupported. The model must be one of: PreTrainedModel, "
+            "TFPreTrainedModel, or FlaxPreTrainedModel"
+        )
+    base_reqs = ["transformers"]
+    base_reqs.extend(_model_engine_type(model))
+    return [_get_pinned_requirement(module) for module in base_reqs]
+
+
+def get_default_conda_env(
+    model,
+) -> Union[None, Dict[str, Union[str, List[Union[str, Dict[str, List[str]]]]]]]:
+    """
+    :return: The default Conda environment for MLflow Models produced with the ``transformers``
+             flavor, based on the model instance framework type of the model to be logged.
+    """
+    return _mlflow_conda_env(additional_pip_deps=get_default_pip_requirements(model))
+
+
+@format_docstring(LOG_MODEL_PARAM_DOCS.format(package_name=FLAVOR_NAME))
```

Comment:
Reminder: Can we annotate all of the public methods in this module with @experimental ?

Reply:
TY for the reminder :)

Reply:
Can we also annotate the `get_default_pip_requirements` and `get_default_conda_env` methods as experimental?

Reply:
Added